packages/webapp/src/components/Form/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,253 @@
+   */
+  allowDecimal?: boolean;
+  /**
+   * The locale to use for number formatting.
+   */
+  locale?: string;
+  /**
+   * Number of decimal places to round on blur. Ignored if used with a step value
+   */
+  roundToDecimalPlaces?: number;
+  /**
+   * The unit to display on right side of input
+   */
+  unit?: ReactNode;
+  /**
+   * Amount to step up or down. Renders a stepper on right side with up and down buttons when step is greater than 0.
+   * If allowDecimal is false, then step is rounded down to the nearest whole number.
+   */
+  step?: number;
+  /**
+   * Max value of input. Only applicable if used with step
+   */
+  max?: number;
+  /**
+   * Min value of input. Only applicable if used with step
+   */
+  min?: number;
+  disabled?: boolean;
+} & InputBaseProps;
+
+const initializeInputValue =
+  (initialValue: NumberInputProps['value'] = '', formatter: Intl.NumberFormat) =>
+  () => {
+    const valueAsNumber = parseFloat(initialValue.toString());
+    const valueString = !Number.isFinite(valueAsNumber) ? '' : formatter.format(valueAsNumber);
+    return {
+      valueString,
+      valueAsNumber,
+    };
+  };
+
+export default function NumberInput({
+  value: propValue = '',
+  onChange,
+  onBlur,
+  useGrouping = true,
+  allowDecimal = true,
+  roundToDecimalPlaces,
+  unit,
+  step,
+  max = Infinity,
+  min = 0,
+  disabled,
+  ...props
+}: NumberInputProps) {
+  const {
+    i18n: { language },
+  } = useTranslation();
+
+  const locale = props.locale || language;
+
+  const formatter = useMemo(() => {
+    const options: Intl.NumberFormatOptions = {
+      useGrouping,
+      maximumFractionDigits: roundToDecimalPlaces ?? 20,
+    };
+
+    if (step) {
+      const decimalPlaces = allowDecimal ? countDecimalPlaces(step) : 0;
+      options.minimumFractionDigits = decimalPlaces;
+      options.maximumFractionDigits = decimalPlaces;
+    }
+
+    try {
+      return new Intl.NumberFormat(locale, options);
+    } catch (error) {
+      return new Intl.NumberFormat(undefined, options);
+    }
+  }, [locale, useGrouping, roundToDecimalPlaces, step, allowDecimal]);
+
+  const { decimalSeparator, thousandsSeparator } = useMemo(() => {
+    let separators = {
+      decimalSeparator: '.',
+      thousandsSeparator: ',',
+    };
+    for (let { type, value } of formatter.formatToParts(11000.2)) {
+      if (type === 'decimal') {
+        separators.decimalSeparator = value;
+      } else if (type === 'group') {
+        separators.thousandsSeparator = value;
+      }
+    }
+    return separators;
+  }, [locale]);
+
+  const [{ valueString, valueAsNumber }, setInputValue] = useState(
+    initializeInputValue(propValue, formatter),
+  );
+  const [isFocused, setIsFocused] = useState(false);
+  const [isEditedAfterFocus, setIsEditedAfterFocus] = useState(false);
+  const initialValueRef = useRef(propValue);
+  const showStepper = typeof step === 'number' && step > 0;
```

Comment:
We don't need to check typeof step but since step is optional it can be undefined so we have to check if it exists. One alternative is to do: `!!step && step > 0`. Or we can also default step to 0 and then it just becomes 
`const showStepper = step > 0`